src/Servers.jl

```diff
@@ -280,14 +280,15 @@ Create a `Transaction` object for each HTTP Request received.
 After `reuse_limit + 1` transactions, signal `final_transaction` to the
 transaction handler.
 """
-function handle_connection(f, c::Connection, reuse_limit, readtimeout)
+function handle_connection(f, c::Connection, server, reuse_limit, readtimeout)
     wait_for_timeout = Ref{Bool}(true)
     if readtimeout > 0
         @async check_readtimeout(c, readtimeout, wait_for_timeout)
     end
     try
         count = 0
-        while isopen(c)
+        # if the connection socket or original server close, we stop taking requests
+        while isopen(c) && isopen(server)
             handle_transaction(f, Transaction(c);
                                final_transaction=(count == reuse_limit))
             count += 1
```

Comment:
Isn't there a problem here if `isopen(c)` is true but `isopen(server)` isn't? i.e. the connection should be closed shouldn't it?

Reply:
It pretty much gets closed immediately by the listenloop once `handle_connection` returns